tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/Integrations/GitHub/GitHubRateLimiter.cs

```diff
@@ -0,0 +1,19 @@
+﻿using ComposableAsync;
+using RateLimiter;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.CheckEnforcer.Integrations.GitHub
+{
+    public static class GitHubRateLimiter
+    {
+        private static TimeLimiter limiter = TimeLimiter.GetFromMaxCountByInterval(1, TimeSpan.FromSeconds(1));
+
+        public async static Task WaitForGitHubCapacityAsync()
+        {
+            await limiter;
```

Comment:
Is it true that this class only helps to serialize the calls? As opposed to actually getting any information from github about rate limiting and trying to balance based on their data? Should we try to feed in any of that information into this?

Reply:
This is based on the best practices on GitHub around throttling traffic to one request per second. At the moment I can only get the rate limit information after an exception occurs which is often too late (we do back off based on the exception, but by that stage we've already hammered the service and the timeouts are long).